analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/LocksReleasedAllPaths.Monitor.TryEnter.cs

```diff
@@ -1,15 +1,17 @@
 ﻿using System;
+using System.Diagnostics;
 using System.Threading;
 
 namespace Monitor_TryEnter
 {
     class Program
     {
         private object obj = new object();
+        private bool condition;
 
-        public void Method1()
+        public void TryEnter_ExitedInElse()
         {
-            if (Monitor.TryEnter(obj)) // Noncompliant
+            if (Monitor.TryEnter(obj)) // Compliant, never exited in this method
```

Comment:
Add a comment about this being a programming error not covered by this rule as in other places:
https://github.com/SonarSource/sonar-dotnet/pull/6052/files#diff-efbbab1580bb1a347cc951d2fcbb9b637389a6b38bc371eff823b2b6b83c4910R78